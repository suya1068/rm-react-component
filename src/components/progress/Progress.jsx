import "./Progress.scss";
import React from "react";
import PropTypes from "prop-types";

export default function Progress() {
    return (
        <div className="u-progress">
            <div className="u-progress__bar">
                <div className="u-progress-status">50%</div>
            </div>
            <div className="u-progress__label">label</div>
        </div>
    );
}

Progress.propTypes = {
    value: PropTypes.number,
    className: PropTypes.shape({
        success: PropTypes.string,
        error: PropTypes.string
    }),
    animation: PropTypes.shape({
        duration: PropTypes.string
    })
};

Progress.defaultProps = {
    value: 0,
    className: {
        success: "u-progress--success",
        error: "u-progress--error"
    },
    animation: {
        duration: "600ms"
    }
};
