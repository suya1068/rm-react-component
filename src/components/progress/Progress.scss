$border-radius: 0.25rem;

.u-progress {
    position: relative;
    display: block;
    height: 1.9em;
    box-sizing: border-box;
    padding: 0;
    margin-bottom: 1.5em;
    max-width: 100%;
    background-color: #fafafa;
    border: none;
    border-radius: $border-radius;

    &__bar {
        display: block;
        height: 1.75em;
        line-height: 1;
        border: none;
        border-radius: $border-radius 0 0 $border-radius;
        background-color: #3f51b5;
        box-shadow: rgba(0, 0, 0, 0.075) 0px 1px 1px inset;
        transition: width, background-color 0.1s ease;
    }

    &__label {
        position: absolute;
        width: 100%;
        left: 0;
        top: 100%;
        right: auto;
        bottom: auto;
        color: rgba(0, 0, 0, 0.87);
        font-size: 1em;
        font-weight: 700;
        margin-top: 0.2em;
        text-shadow: none;
        text-align: center;
        transition: color 0.4s ease;
    }

    &-status {
        position: absolute;
        width: auto;
        left: auto;
        top: 50%;
        bottom: auto;
        right: 0.5em;
        transform: translateY(-50%);
        font-size: 0.95rem;
        color: rgba(255, 255, 255, 0.7);
        white-space: nowrap;
        text-shadow: none;
        font-weight: 700;
        text-align: left;
    }
}
