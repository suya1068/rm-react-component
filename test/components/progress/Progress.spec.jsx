import React from "react";
import Progress from "src/components/progress/Progress";

describe("# Progress Component", () => {
    it("render <Progres />", () => {
        const tree = shallow(<Progress />);
        expect(tree).toMatchSnapshot();
    });
});
