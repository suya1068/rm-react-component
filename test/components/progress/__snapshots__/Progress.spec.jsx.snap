// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`# Progress Component render <Progres /> 1`] = `
<div
  className="u-progress"
>
  <div
    className="u-progress__bar"
  >
    <div
      className="u-progress-status"
    >
      50%
    </div>
  </div>
  <div
    className="u-progress__label"
  >
    label
  </div>
</div>
`;
